pkg/controller/constrainttemplate/constrainttemplate_controller.go

```diff
@@ -904,16 +912,16 @@ func (r *ReconcileConstraintTemplate) updateTemplateWithBlockVAPBGenerationAnnot
 		// otherwise update the annotation with the current time + wait time. This prevents clock skew from preventing generation on task reschedule.
 		if t.Before(currentTime.Add(time.Duration(*constraint.DefaultWaitForVAPBGeneration) * time.Second)) {
 			if t.Before(currentTime) {
-				ct.Annotations[constraint.VAPBGenerationAnnotation] = constraint.VAPBGenerationUnblocked
-				return noRequeue, r.Update(ctx, ct)
+				latestCT.Annotations[constraint.VAPBGenerationAnnotation] = constraint.VAPBGenerationUnblocked
+				return noRequeue, r.Update(ctx, latestCT)
 			}
```

Comment:
Would it be more idiomatic to handle the race condition by instead of doing a GET on l897 for the latest template, just wrap this Update logic in a backoff retry with GET + Update to check for conflicts